### DISCUSSION TITLE
Line and Marker animation fo real-time graph

### DISCUSSION THREAD

**Discussion by GeorgiyShur** [patrykandpatrick/vico#885] [Questions] [answered]
### Question

Hi, I'm working on a real-time graph that displays changes in "current" value. So far I'm super impressed with Vico and I was able to implement almost every detail from the requirements. 

Unfortunately I wasn't able to find a way to implement smooth animations for value change for all the use cases.

It works perfectly when I'm updating the existing data point, the marker and the latest value animate correctly and it looks awesome:

https://github.com/user-attachments/assets/97c81c9a-4fb4-4cb7-9468-2ca2eb8ed999

But if I'm adding a new value, it always animates from zero, and as far as I can see, there is no way to turn off this animation without turning all the animations off:

https://github.com/user-attachments/assets/3e89e656-b3cd-457c-80a4-eed2a43d1609

My questions are:
1. Is there any way to implement the custom animation for the marker and the line, so it travels from the previous value?
2. Is there any chance that you guys will support something like this in the future? Seems like for real-time graphs it's a pretty common use case.

P.S. I found [this discussion](https://github.com/patrykandpatrick/vico/discussions/563), but it doesn't seem to answer my questions fully.

### Vico version(s)

2.0.0-alpha.28

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment, marked as answer]
Hello @GeorgiyShur.
>as far as I can see, there is no way to turn off this animation without turning all the animations off

It is possible. You can do it by creating your own `CartesianLayerDrawingModelInterpolator`. You can use the [default implementation](https://github.com/patrykandpatrick/vico/blob/master/vico/core/src/main/java/com/patrykandpatrick/vico/core/common/data/DefaultCartesianLayerDrawingModelInterpolator.kt) as a starting point. You just need a minor change in `updateTransformationMap`. 

We plan on adding a parameter for disabling the animation from zero.

>Is there any way to implement the custom animation for the marker and the line, so it travels from the previous value?

Would you like both _x_ and _y_ axis values, or just _y_ axis value?

**Reply by GeorgiyShur** [reply]
Thanks a lot for a prompt response :)

> Would you like both x and y axis values, or just y axis value?

For this particular use case, just y axis should be enough, because the x position of "current value" won't change, but I could see value in both :)

I'll try your suggestion of creating my own interpolator and come back with the result.

Thanks for your help and doing great work with this library!

**Reply by GeorgiyShur** [reply]
Thanks again for your suggestion, I was able to implement the desired behavior by backing up the `TransformationModel` old value with the previous one in `updateTransformationMap` function.

**Reply by Gowsky** [reply]
I'm glad you have found the solution. Cheers!

**Reply by spectrl** [reply]
I'm looking for the same functionality, did this new parameter make it in yet? @Gowsky 

Otherwise I'll attempt the custom interpolator 🤞

**Reply by udyr-woo** [reply]
> Thanks again for your suggestion, I was able to implement the desired behavior by backing up the `TransformationModel` old value with the previous one in `updateTransformationMap` function.

@GeorgiyShur Can I get your compose source? I can't build real time chart like yours

**Reply by Gowsky** [reply]
@udyr-woo, based on #996 the `Flow` comes from a `ViewModel`; therefore you should create and interact with the `CartesianChartModelProducer` there, not in Compose. You should then pass the producer to your composable function, which should have only a `CartesianChartModelProducer` parameter.
